### PR TITLE
Fixed test:whenLoadAllIsUsed_thenNearCacheIsInvalidated_onDataAdapter

### PR DESCRIPTION
**Pull request by ahmetmircik**

### CODE REVIEW

**Comment by jerrinot** [comment]
@ahmetmircik: can you please add some description? what was the problem and what's the idea of the fix.
it's not immediately clear to me.

**Comment by Donnerbart** [comment]
The motivation is to get the `whenLoadAllIsUsed_thenNearCacheIsInvalidated_onDataAdapter()` test running, which was ignored on the Near Cache test backport.

User story: When I have a `MapLoader` which loads all keys, then the Near Cache on my client should be invalidated.

**Comment by jerrinot** [comment]
@Donnerbart: ok. I assume you and @tombujok can review it

File: hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/BasicRecordStoreLoader.java [hunk 1]
```diff
@@ -74,8 +74,8 @@
     }
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/BasicRecordStoreLoader.java]: Due to this refactoring where the methods were moved to inner class it's super tough to read what has changed. 
Could we revert this refactoring and then have a separate commit where the methods are moved...

**Comment by tombujok** [comment]
Let's discuss this fix later. I'm moving the milestone to 3.8.4 for now. it would be good to have sth that does not pollute the all layers with callerUuid.

**Comment by mdogan** [comment]
@ahmetmircik: What is the latest state here? PR's milestone is 3.9 but commit goes to `hazelcast:maintenance-3.x` branch.

**Comment by ahmetmircik** [comment]
will send a different PR

**Pull request closed without merging**